test/e2e/specs/realrp/vnetpeering.go

```diff
@@ -0,0 +1,125 @@
+package realrp
+
+import (
+	"context"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-06-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-05-01/resources"
+
+	"github.com/openshift/openshift-azure/pkg/api/2018-09-30-preview/api"
+	"github.com/openshift/openshift-azure/pkg/fakerp/client"
+	"github.com/openshift/openshift-azure/test/clients/azure"
+	tlog "github.com/openshift/openshift-azure/test/util/log"
+)
+
+var _ = Describe("Peer Vnet tests [Vnet][Real][LongRunning]", func() {
+	var (
+		ctx          = context.Background()
+		cli          *azure.Client
+		rg           = os.Getenv("RESOURCEGROUP")
+		vnetPeerName = "vnetPeer"
+		region       = os.Getenv("AZURE_REGION")
+		clientID     = os.Getenv("AZURE_CLIENT_ID")
+		clientSecret = os.Getenv("AZURE_CLIENT_SECRET")
+	)
+
+	BeforeEach(func() {
+		var err error
+		cli, err = azure.NewClientFromEnvironment(false)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(region).NotTo(BeEmpty())
+		Expect(rg).NotTo(BeEmpty())
+	})
+
+	AfterEach(func() {
+		result, _ := cli.Groups.Delete(ctx, rg)
```

Comment:
Instead of just deleting the resource group, I think it's more appropriate to first Delete the OSA resource, then the resource group, similarly to how you first create the resource group, then the OSA resource. There may be other actions triggered in the OSA RP backend apart from resouces in the current resource group?